src/DataTypes/IDataType.h

```diff
@@ -75,6 +75,8 @@ class IDataType : private boost::noncopyable, public std::enable_shared_from_thi
 
     /// Name of data type family (example: FixedString, Array).
     virtual const char * getFamilyName() const = 0;
+    /// Name of corresponding data type in MySQL (exampe: Bigint, Blob, etc)
```

Comment:
(For consistency, it would be great to have `getFamilyName()` also return a String. But that's unrelated and fine to be done in a separate PR.)

Reply:
It's ok for `getFamilyName` to return a reference to a statically allocated string (const char *), because the family name is constant for the class.